WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -246,23 +255,25 @@ def post_results
         if top_three.empty?
           return render html: "<div class='container'><div class='alert alert-warning'>Nobody competed in event: #{event.id}</div></div>".html_safe
         else
-          first_result = top_three.first
+          results_by_place = top_three.group_by(&:pos)
+          winners = results_by_place[1]
 
-          title = "#{first_result.personName} wins #{comp.name}, in #{comp.cityName}, #{comp.country.name}"
+          title = "#{people_to_sentence(winners, link: false)} #{winners.length  > 1 ? "win" : "wins"} " \
+                  "#{comp.name}, in #{comp.cityName}, #{comp.country.name}"
 
-          body = "[#{first_result.personName}](#{person_url first_result.personId})"
-          body += " won the [#{comp.name}](#{competition_url(comp)})"
-          body += " #{pretty_print_result(first_result)}"
+          body = "#{people_to_sentence(winners, link: true)} won the [#{comp.name}](#{competition_url(comp)})" \
+                 " #{pretty_print_result(winners.first)}" # If there are more winners then their results are the same.
           body += " in the #{event.name} event" if event.id != "333"
-
-          if top_three.length > 1
-            body += ". [#{top_three.second.personName}](#{person_url top_three.second.personId}) finished second (#{pretty_print_result(top_three.second, short: true)})"
-            if top_three.length > 2
-              body += " and [#{top_three.third.personName}](#{person_url top_three.third.personId}) finished third (#{pretty_print_result(top_three.third, short: true)})"
-            end
+          body += "."
+          if results_by_place[2]
+            body += " #{people_to_sentence(results_by_place[2], link: true)} finished second (#{pretty_print_result(top_three.second, short: true)})"
+            body += results_by_place[3] ? " and" : "."
           end
-
-          body += ".\n\n"
+          if results_by_place[3]
```

Comment:
Out of curiosity, can this happen if the previous `if results_by_place[2]` did not happen? Or did you just move this out here to reduce indentation?

Reply:
This can surely be the case, you can even see it in the test where two people take the first place, so there are two `1st` places, one `3rd` place, but no `2nd` places at all.

Reply:
Oh, right! Silly ties :)